core/arch/arm/kernel/link.mk

```diff
@@ -102,14 +102,13 @@ define update-buildcount
 		expr 0`cat $(1)` + 1 >$(1); \
 	fi
 endef
-VERSION_STR = `git describe --always --dirty 2>/dev/null || echo Unknown`
 DATE_STR = `date -u`
 BUILD_COUNT_STR = `cat $(link-out-dir)/.buildcount`
 define gen-version-o
 	$(call update-buildcount,$(link-out-dir)/.buildcount)
 	@$(cmd-echo-silent) '  GEN     $(link-out-dir)/version.o'
 	$(q)echo -e "const char core_v_str[] =" \
-		"\"$(VERSION_STR) \"" \
+		"\"$(CFG_TEE_IMPL_VERSION) \"" \
```

Comment:
Yes, quoting is wrong. `CFG_TEE_IMPLENTATION` is defined in `config.mk` and starts/ends with double quotes. I think `CFG_*` variables should *not* be quoted because you simply don't know how you are supposed to quote them. It depends on which language will use them later (Make? C?).
So my suggestion is to modify `config.mk` as follows:
```
CFG_TEE_IMPL_VERSION ?= $(shell git describe --dirty=-dev 2>/dev/null || echo Unknown)
```
And keep `link.mk` as above.
Note: if we want to quote strings in C files we can always do:
```
#define TO_STR(x) _TO_STR(x)
#define _TO_STR(x) #x
const char foo[] = TO_STR(CFG_TEE_IMPL_VERSION);
```